libs/data-access/api/src/lib/trpc/routers/settings.router.ts

```diff
@@ -18,6 +18,22 @@ export const settingsRouter = t.router({
 			})
 		)
 		.mutation(async ({ ctx, input }) => {
+			const currentSettings = await database.studentSettings.findUnique({
+				where: {
+					username: ctx.user.name
+				}
+			});
+
+			if (currentSettings?.hasLearningDiary !== input.settings.hasLearningDiary) {
+				await database.eventLog.create({
+					data: {
+						type: "LTB_TOGGLE",
+						payload: { enabled: input.settings.hasLearningDiary },
+						username: ctx.user.name
+					}
+				});
+			}
+
 			return await database.studentSettings.upsert({
```

Comment:
beide statements sollten zusammen in eine transaction. ansonsten wird das LTB_TOGGLE eingetragen, selbst dann, wenn danach das upsert fehlschlägt.

Reply:
Wenn das in einem passiert, wird aber auch ein Toggle Event aufgerufen, wenn man Statistiken aktiviert. Dann wird der aktuelle Status des LTB (ausgeschaltet) noch einmal als Event in den Log geschrieben. Wenn das kein Problem ist, kann ich das in eins tun. Sonst nicht.